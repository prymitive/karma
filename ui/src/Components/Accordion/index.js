import React from "react";

import Collapsible from "react-collapsible";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faChevronUp } from "@fortawesome/free-solid-svg-icons/faChevronUp";
import { faChevronDown } from "@fortawesome/free-solid-svg-icons/faChevronDown";

const Trigger = ({ text, isOpen }) => (
  <div className="d-flex flex-row justify-content-between">
    <div>{text}</div>
    <div>
      <FontAwesomeIcon
        icon={isOpen ? faChevronDown : faChevronUp}
        className="text-muted"
      />
    </div>
  </div>
);

const Accordion = ({ text, content, extraProps }) => (
  <Collapsible
    triggerTagName="div"
    transitionTime={50}
    className="card"
    openedClassName="card"
    triggerClassName="card-header cursor-pointer"
    triggerOpenedClassName="card-header cursor-pointer bg-light"
    contentOuterClassName="collapse show"
    contentInnerClassName="card-body my-2"
    {...extraProps}
    trigger={<Trigger text={text} isOpen={false} />}
    triggerWhenOpen={<Trigger text={text} isOpen={true} />}
  >
    {content}
  </Collapsible>
);

export { Accordion };
