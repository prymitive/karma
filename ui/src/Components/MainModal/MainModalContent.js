import React, { useState } from "react";
import PropTypes from "prop-types";

import { useObserver } from "mobx-react-lite";

import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { Tab } from "Components/Modal/Tab";
import { Configuration } from "./Configuration";
import { Help } from "./Help";

const TabNames = Object.freeze({
  Configuration: "configuration",
  Help: "help",
});

const MainModalContent = ({
  alertStore,
  settingsStore,
  onHide,
  openTab,
  expandAllOptions,
}) => {
  const [tab, setTab] = useState(openTab);

  return useObserver(() => (
    <React.Fragment>
      <div className="modal-header py-2">
        <nav className="nav nav-pills nav-justified w-100">
          <Tab
            title="Configuration"
            active={tab === TabNames.Configuration}
            onClick={() => setTab(TabNames.Configuration)}
          />
          <Tab
            title="Help"
            active={tab === TabNames.Help}
            onClick={() => setTab(TabNames.Help)}
          />
          <button type="button" className="close" onClick={onHide}>
            <span>&times;</span>
          </button>
        </nav>
      </div>
      <div className="modal-body">
        {tab === TabNames.Help ? (
          <Help defaultIsOpen={expandAllOptions} />
        ) : null}
        {tab === TabNames.Configuration ? (
          <Configuration
            settingsStore={settingsStore}
            defaultIsOpen={expandAllOptions}
          />
        ) : null}
      </div>
      <div className="modal-footer">
        {alertStore.info.authentication.enabled && (
          <span className="text-muted mr-2">
            Username: {alertStore.info.authentication.username}
          </span>
        )}
        <span className="text-muted">Version: {alertStore.info.version}</span>
      </div>
    </React.Fragment>
  ));
};
MainModalContent.propTypes = {
  alertStore: PropTypes.instanceOf(AlertStore).isRequired,
  settingsStore: PropTypes.instanceOf(Settings).isRequired,
  onHide: PropTypes.func.isRequired,
  openTab: PropTypes.oneOf(Object.values(TabNames)),
  expandAllOptions: PropTypes.bool.isRequired,
};
MainModalContent.defaultProps = {
  openTab: TabNames.Configuration,
};

export { MainModalContent, TabNames };
