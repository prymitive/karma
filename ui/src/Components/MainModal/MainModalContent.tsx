import { FC, useState } from "react";

import { Observer } from "mobx-react-lite";

import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { Tab } from "Components/Modal/Tab";
import { Configuration } from "./Configuration";
import { Help } from "./Help";

export type OpenTabT = "configuration" | "help";

const MainModalContent: FC<{
  alertStore: AlertStore;
  settingsStore: Settings;
  onHide: () => void;
  openTab?: OpenTabT;
  expandAllOptions: boolean;
}> = ({
  alertStore,
  settingsStore,
  onHide,
  openTab = "configuration",
  expandAllOptions,
}) => {
  const [tab, setTab] = useState<OpenTabT>(openTab);

  return (
    <>
      <div className="modal-header py-2">
        <nav className="nav nav-pills nav-justified w-100">
          <Tab
            title="Configuration"
            active={tab === "configuration"}
            onClick={() => setTab("configuration")}
          />
          <Tab
            title="Help"
            active={tab === "help"}
            onClick={() => setTab("help")}
          />
          <button
            type="button"
            className="btn-close my-auto"
            onClick={onHide}
          ></button>
        </nav>
      </div>
      <div className="modal-body">
        {tab === "help" ? <Help defaultIsOpen={expandAllOptions} /> : null}
        {tab === "configuration" ? (
          <Configuration
            settingsStore={settingsStore}
            defaultIsOpen={expandAllOptions}
          />
        ) : null}
      </div>
      <div className="modal-footer">
        <Observer>
          {() =>
            alertStore.info.authentication.enabled ? (
              <span className="text-muted me-2">
                Username: {alertStore.info.authentication.username}
              </span>
            ) : null
          }
        </Observer>
        <Observer>
          {() => (
            <span className="text-muted">
              Version: {alertStore.info.version}
            </span>
          )}
        </Observer>
      </div>
    </>
  );
};

export { MainModalContent };
