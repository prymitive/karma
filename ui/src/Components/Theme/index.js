import React, { useEffect } from "react";
import ReactDOM from "react-dom";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSun } from "@fortawesome/free-solid-svg-icons/faSun";

const DarkTheme = React.lazy(() => import("Styles/DarkTheme"));
const LightTheme = React.lazy(() => import("Styles/LightTheme"));

const Placeholder = () => {
  return ReactDOM.createPortal(
    <div
      style={{
        zIndex: 2000,
        backgroundColor: "#455a64",
        position: "fixed",
        top: 0,
        left: 0,
        width: "100vw",
        height: "100vh",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      <FontAwesomeIcon
        icon={faSun}
        size="lg"
        spin
        style={{
          color: "#5e7a88",
          fontSize: "14rem",
        }}
      />
    </div>,
    document.body
  );
};

const ThemeContext = React.createContext();

const BodyTheme = () => {
  const context = React.useContext(ThemeContext);

  useEffect(() => {
    document.body.classList.toggle("theme-dark", context.isDark);
    document.body.classList.toggle("theme-light", !context.isDark);
  }, [context.isDark]);

  return (
    <React.Suspense fallback={<Placeholder />}>
      {context.isDark ? <DarkTheme /> : <LightTheme />}
    </React.Suspense>
  );
};

export { BodyTheme, ThemeContext };
