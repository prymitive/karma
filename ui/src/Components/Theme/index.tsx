import React, { FC, useEffect } from "react";
import ReactDOM from "react-dom";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSun } from "@fortawesome/free-solid-svg-icons/faSun";

import { Styles } from "react-select";

const DarkTheme = React.lazy(() => import("Styles/DarkTheme"));
const LightTheme = React.lazy(() => import("Styles/LightTheme"));

const Placeholder = () => {
  return ReactDOM.createPortal(
    <div
      style={{
        zIndex: 2000,
        backgroundColor: "#455a64",
        position: "fixed",
        top: 0,
        left: 0,
        width: "100vw",
        height: "100vh",
        display: "flex",
        alignItems: "center",
        justifyContent: "center",
        flexDirection: "column",
      }}
    >
      <FontAwesomeIcon
        icon={faSun}
        size="lg"
        spin
        style={{
          color: "#5e7a88",
          fontSize: "14rem",
        }}
      />
    </div>,
    document.body
  );
};

export interface ThemeCtx {
  isDark: boolean;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  reactSelectStyles: Styles<any, any>;
  animations: {
    duration: number;
  };
}

const ThemeContext = React.createContext({
  isDark: false,
  reactSelectStyles: {},
  animations: { duration: 1000 },
} as ThemeCtx);

const BodyTheme: FC = () => {
  const context = React.useContext(ThemeContext);

  useEffect(() => {
    document.body.classList.toggle("theme-dark", context.isDark);
    document.body.classList.toggle("theme-light", !context.isDark);
  }, [context.isDark]);

  return (
    <React.Suspense fallback={<Placeholder />}>
      {context.isDark ? <DarkTheme /> : <LightTheme />}
    </React.Suspense>
  );
};

export { BodyTheme, ThemeContext };
