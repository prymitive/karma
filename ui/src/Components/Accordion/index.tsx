import React, { FunctionComponent, ReactNode, useState } from "react";

import { ToggleIcon } from "Components/ToggleIcon";

const Trigger: FunctionComponent<{ text: string; isOpen: boolean }> = ({
  text,
  isOpen,
}) => (
  <div className="d-flex flex-row justify-content-between">
    <div>{text}</div>
    <div>
      <ToggleIcon isOpen={isOpen} className="text-muted" />
    </div>
  </div>
);

const Accordion: FunctionComponent<{
  text: string;
  content: ReactNode;
  defaultIsOpen?: boolean;
}> = ({ text, content, defaultIsOpen }) => {
  const [isOpen, setIsOpen] = useState(defaultIsOpen || false);

  return (
    <div className="accordion card">
      <div
        className={`card-header cursor-pointer ${isOpen ? "active" : ""}`}
        onClick={() => setIsOpen(!isOpen)}
      >
        <Trigger text={text} isOpen={isOpen} />
      </div>
      <div className={isOpen ? "card-body my-2" : ""}>{isOpen && content}</div>
    </div>
  );
};

export { Accordion };
