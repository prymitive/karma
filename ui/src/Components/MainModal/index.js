import React, { useState, useCallback } from "react";
import PropTypes from "prop-types";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCog } from "@fortawesome/free-solid-svg-icons/faCog";
import { faSpinner } from "@fortawesome/free-solid-svg-icons/faSpinner";

import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { TooltipWrapper } from "Components/TooltipWrapper";
import { Modal } from "Components/Modal";

// https://github.com/facebook/react/issues/14603
const MainModalContent = React.lazy(() =>
  import("./MainModalContent").then((module) => ({
    default: module.MainModalContent,
  }))
);

const MainModal = ({ alertStore, settingsStore }) => {
  const [isVisible, setIsVisible] = useState(false);

  const toggle = useCallback(() => setIsVisible(!isVisible), [isVisible]);

  return (
    <React.Fragment>
      <li
        className={`nav-item components-navbar-button ${
          isVisible ? "border-info" : ""
        }`}
      >
        <TooltipWrapper title="Settings">
          <span
            id="components-settings"
            className="nav-link cursor-pointer"
            onClick={toggle}
          >
            <FontAwesomeIcon icon={faCog} />
          </span>
        </TooltipWrapper>
      </li>
      <Modal isOpen={isVisible} toggleOpen={toggle}>
        <React.Suspense
          fallback={
            <h1 className="display-1 text-placeholder p-5 m-auto">
              <FontAwesomeIcon icon={faSpinner} size="lg" spin />
            </h1>
          }
        >
          <MainModalContent
            alertStore={alertStore}
            settingsStore={settingsStore}
            onHide={() => setIsVisible(false)}
            isVisible={isVisible}
            expandAllOptions={false}
          />
        </React.Suspense>
      </Modal>
    </React.Fragment>
  );
};
MainModal.propTypes = {
  alertStore: PropTypes.instanceOf(AlertStore).isRequired,
  settingsStore: PropTypes.instanceOf(Settings).isRequired,
};

export { MainModal };
