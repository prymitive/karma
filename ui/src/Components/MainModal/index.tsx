import React, { FC, useState, useCallback } from "react";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSlidersH } from "@fortawesome/free-solid-svg-icons/faSlidersH";
import { faSpinner } from "@fortawesome/free-solid-svg-icons/faSpinner";

import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { TooltipWrapper } from "Components/TooltipWrapper";
import { Modal } from "Components/Modal";

// https://github.com/facebook/react/issues/14603
const MainModalContent = React.lazy(() =>
  import("./MainModalContent").then((module) => ({
    default: module.MainModalContent,
  }))
);

const MainModal: FC<{
  alertStore: AlertStore;
  settingsStore: Settings;
}> = ({ alertStore, settingsStore }) => {
  const [isVisible, setIsVisible] = useState<boolean>(false);

  const toggle = useCallback(() => setIsVisible(!isVisible), [isVisible]);

  return (
    <>
      <li
        className={`nav-item components-navbar-button ${
          isVisible ? "border-info" : ""
        } ml-auto`}
      >
        <TooltipWrapper title="Settings">
          <span
            id="components-settings"
            className="nav-link cursor-pointer"
            onClick={toggle}
          >
            <FontAwesomeIcon icon={faSlidersH} fixedWidth />
          </span>
        </TooltipWrapper>
      </li>
      <Modal isOpen={isVisible} toggleOpen={toggle}>
        <React.Suspense
          fallback={
            <h1 className="display-1 text-placeholder p-5 m-auto">
              <FontAwesomeIcon icon={faSpinner} size="lg" spin />
            </h1>
          }
        >
          <MainModalContent
            alertStore={alertStore}
            settingsStore={settingsStore}
            onHide={() => setIsVisible(false)}
            expandAllOptions={false}
          />
        </React.Suspense>
      </Modal>
    </>
  );
};

export { MainModal };
