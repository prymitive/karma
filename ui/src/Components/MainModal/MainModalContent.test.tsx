import { mount } from "enzyme";

import fetchMock from "fetch-mock";

import toDiffableHtml from "diffable-html";

import { MockThemeContext } from "__fixtures__/Theme";
import { AlertStore } from "Stores/AlertStore";
import { Settings } from "Stores/Settings";
import { ThemeContext } from "Components/Theme";
import { MainModalContent } from "./MainModalContent";

let alertStore: AlertStore;
let settingsStore: Settings;
const onHide = jest.fn();

beforeEach(() => {
  alertStore = new AlertStore([]);
  settingsStore = new Settings(null);
  onHide.mockClear();
  fetchMock.reset();
  fetchMock.mock("*", {
    body: JSON.stringify([]),
  });
});

afterEach(() => {
  jest.restoreAllMocks();
  fetchMock.reset();
});

const Wrapped = (component: any) => (
  <ThemeContext.Provider value={MockThemeContext}>
    {component}
  </ThemeContext.Provider>
);

const FakeModal = () => {
  return mount(
    Wrapped(
      <MainModalContent
        alertStore={alertStore}
        settingsStore={settingsStore}
        onHide={onHide}
        expandAllOptions={true}
      />
    )
  );
};

const ValidateSetTab = (title: string) => {
  const component = FakeModal();

  const tab = component.find({ title: title });
  tab.simulate("click");
  expect(component.find(".nav-link.active").text()).toBe(title);
};

describe("<MainModalContent />", () => {
  it("matches snapshot", () => {
    // we have multiple fragments and enzyme only renders the first one
    // in html() and text(), debug() would work but it's noisy
    // https://github.com/airbnb/enzyme/issues/1213
    const tree = mount(
      <span>
        {Wrapped(
          <MainModalContent
            alertStore={alertStore}
            settingsStore={settingsStore}
            onHide={onHide}
            expandAllOptions={true}
          />
        )}
      </span>
    );
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("shows 'Configuration' tab by default", () => {
    const tree = FakeModal();
    const activeTab = tree.find(".nav-link.active");
    expect(activeTab.text()).toBe("Configuration");
  });

  // modal makes it tricky to verify re-rendered content, so only check if we
  // update the store for now
  it("calls setTab('configuration') after clicking on the 'Configuration' tab", () => {
    ValidateSetTab("Configuration");
  });

  it("calls setTab('help') after clicking on the 'Help' tab", () => {
    ValidateSetTab("Help");
  });

  it("shows username when alertStore.info.authentication.enabled=true", () => {
    alertStore.info.setAuthentication(true, "me@example.com");
    const tree = mount(
      <span>
        {Wrapped(
          <MainModalContent
            alertStore={alertStore}
            settingsStore={settingsStore}
            onHide={onHide}
            expandAllOptions={true}
          />
        )}
      </span>
    );
    expect(tree.text()).toMatch(/Username: me@example.com/);
  });
});
