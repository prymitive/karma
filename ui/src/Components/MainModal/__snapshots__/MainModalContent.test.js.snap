// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainModalContent /> matches snapshot 1`] = `
"
<span>
  <div class=\\"modal-header py-2\\">
    <nav class=\\"nav nav-pills nav-justified w-100\\">
      <span class=\\"nav-item nav-link cursor-pointer active\\">
        Configuration
      </span>
      <span class=\\"nav-item nav-link cursor-pointer text-primary\\">
        Help
      </span>
      <button type=\\"button\\"
              class=\\"close\\"
      >
        <span>
          Ã—
        </span>
      </button>
    </nav>
  </div>
  <div class=\\"modal-body\\">
    <form class=\\"px-3\\">
      <div class=\\"form-group text-center\\">
        <label class=\\"mb-4 font-weight-bold\\">
          Refresh interval
        </label>
        <div aria-disabled=\\"false\\"
             class=\\"input-range\\"
        >
          <span class=\\"input-range__label input-range__label--min\\">
            <span class=\\"input-range__label-container\\">
              10s
            </span>
          </span>
          <div class=\\"input-range__track input-range__track--background\\">
            <div style=\\"left: 0%; width: 18.181818181818183%;\\"
                 class=\\"input-range__track input-range__track--active\\"
            >
            </div>
            <span class=\\"input-range__slider-container\\"
                  style=\\"position: absolute; left: 18.181818181818183%;\\"
            >
              <span class=\\"input-range__label input-range__label--value\\">
                <span class=\\"input-range__label-container\\">
                  30s
                </span>
              </span>
              <div aria-valuemax=\\"120\\"
                   aria-valuemin=\\"10\\"
                   aria-valuenow=\\"30\\"
                   class=\\"input-range__slider\\"
                   draggable=\\"false\\"
                   role=\\"slider\\"
                   tabindex=\\"0\\"
              >
              </div>
            </span>
          </div>
          <span class=\\"input-range__label input-range__label--max\\">
            <span class=\\"input-range__label-container\\">
              120s
            </span>
          </span>
        </div>
      </div>
      <div class=\\"mt-5\\">
      </div>
      <div class=\\"form-group\\">
        <div class=\\"text-center\\">
          <label class=\\"mb-2 font-weight-bold\\">
            Filter bar configuration
          </label>
        </div>
        <div class=\\"form-check form-check-inline\\">
          <span class=\\"custom-control custom-switch\\">
            <input id=\\"configuration-autohide\\"
                   class=\\"custom-control-input\\"
                   type=\\"checkbox\\"
                   value
                   checked
            >
            <label class=\\"custom-control-label cursor-pointer mr-3\\"
                   for=\\"configuration-autohide\\"
            >
              Hide filter bar when idle
            </label>
          </span>
        </div>
      </div>
      <div class=\\"mt-5\\">
      </div>
      <div class=\\"form-group\\">
        <div class=\\"text-center\\">
          <label class=\\"mb-2 font-weight-bold\\">
            Alert group titlebar configuration
          </label>
        </div>
        <div class=\\"form-check form-check-inline\\">
          <span class=\\"custom-control custom-switch\\">
            <input id=\\"configuration-colortitlebar\\"
                   class=\\"custom-control-input\\"
                   type=\\"checkbox\\"
                   value
            >
            <label class=\\"custom-control-label cursor-pointer mr-3\\"
                   for=\\"configuration-colortitlebar\\"
            >
              Color group titlebar
            </label>
          </span>
        </div>
      </div>
      <div class=\\"mt-5\\">
      </div>
      <div class=\\"form-group text-center\\">
        <label class=\\"mb-4 font-weight-bold\\">
          Minimal alert group width
        </label>
        <div aria-disabled=\\"false\\"
             class=\\"input-range\\"
        >
          <span class=\\"input-range__label input-range__label--min\\">
            <span class=\\"input-range__label-container\\">
              300
            </span>
          </span>
          <div class=\\"input-range__track input-range__track--background\\">
            <div style=\\"left: 0%; width: 24%;\\"
                 class=\\"input-range__track input-range__track--active\\"
            >
            </div>
            <span class=\\"input-range__slider-container\\"
                  style=\\"position: absolute; left: 24%;\\"
            >
              <span class=\\"input-range__label input-range__label--value\\">
                <span class=\\"input-range__label-container\\">
                  420
                </span>
              </span>
              <div aria-valuemax=\\"800\\"
                   aria-valuemin=\\"300\\"
                   aria-valuenow=\\"420\\"
                   class=\\"input-range__slider\\"
                   draggable=\\"false\\"
                   role=\\"slider\\"
                   tabindex=\\"0\\"
              >
              </div>
            </span>
          </div>
          <span class=\\"input-range__label input-range__label--max\\">
            <span class=\\"input-range__label-container\\">
              800
            </span>
          </span>
        </div>
      </div>
      <div class=\\"mt-5\\">
      </div>
      <div class=\\"form-group text-center\\">
        <label class=\\"mb-4 font-weight-bold\\">
          Default number of alerts to show per group
        </label>
        <div aria-disabled=\\"false\\"
             class=\\"input-range\\"
        >
          <span class=\\"input-range__label input-range__label--min\\">
            <span class=\\"input-range__label-container\\">
              1
            </span>
          </span>
          <div class=\\"input-range__track input-range__track--background\\">
            <div style=\\"left: 0%; width: 44.44444444444444%;\\"
                 class=\\"input-range__track input-range__track--active\\"
            >
            </div>
            <span class=\\"input-range__slider-container\\"
                  style=\\"position: absolute; left: 44.44444444444444%;\\"
            >
              <span class=\\"input-range__label input-range__label--value\\">
                <span class=\\"input-range__label-container\\">
                  5
                </span>
              </span>
              <div aria-valuemax=\\"10\\"
                   aria-valuemin=\\"1\\"
                   aria-valuenow=\\"5\\"
                   class=\\"input-range__slider\\"
                   draggable=\\"false\\"
                   role=\\"slider\\"
                   tabindex=\\"0\\"
              >
              </div>
            </span>
          </div>
          <span class=\\"input-range__label input-range__label--max\\">
            <span class=\\"input-range__label-container\\">
              10
            </span>
          </span>
        </div>
      </div>
      <div class=\\"mt-5\\">
      </div>
      <div class=\\"form-group\\">
        <div class=\\"text-center\\">
          <label class=\\"mb-4 font-weight-bold\\">
            Default alert group display
          </label>
        </div>
        <div class=\\" css-2b097c-container\\">
          <div class=\\"react-select__control css-au8pbo-control\\">
            <div class=\\"react-select__value-container react-select__value-container--has-value css-pb81dw\\">
              <div class=\\"react-select__single-value css-1uccc91-singleValue\\">
                Collapse on mobile
              </div>
              <div class=\\"css-1g6gooi\\">
                <div class=\\"react-select__input\\"
                     style=\\"display: inline-block;\\"
                >
                  <input autocapitalize=\\"none\\"
                         autocomplete=\\"off\\"
                         autocorrect=\\"off\\"
                         id=\\"react-select-configuration-collapse-input\\"
                         spellcheck=\\"false\\"
                         tabindex=\\"0\\"
                         type=\\"text\\"
                         aria-autocomplete=\\"list\\"
                         style=\\"box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;\\"
                         value
                  >
                  <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
                  </div>
                </div>
              </div>
            </div>
            <div class=\\"react-select__indicators css-mik995\\">
              <span class=\\"react-select__indicator-separator css-1okebmr-indicatorSeparator\\">
              </span>
              <div aria-hidden=\\"true\\"
                   class=\\"react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer\\"
              >
                <svg height=\\"20\\"
                     width=\\"20\\"
                     viewbox=\\"0 0 20 20\\"
                     aria-hidden=\\"true\\"
                     focusable=\\"false\\"
                     class=\\"css-19bqh2r\\"
                >
                  <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
                  </path>
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class=\\"mt-5\\">
      </div>
      <div class=\\"form-group\\">
        <div class=\\"text-center\\">
          <label class=\\"mb-2 font-weight-bold\\">
            Grid sort order
          </label>
        </div>
        <div class=\\"d-flex flex-fill flex-lg-row flex-column justify-content-between\\">
          <div class=\\"flex-shrink-0 flex-grow-1 flex-basis-auto\\">
            <div class=\\" css-2b097c-container\\">
              <div class=\\"react-select__control css-au8pbo-control\\">
                <div class=\\"react-select__value-container react-select__value-container--has-value css-pb81dw\\">
                  <div class=\\"react-select__single-value css-1uccc91-singleValue\\">
                    Use defaults from karma config file
                  </div>
                  <div class=\\"css-1g6gooi\\">
                    <div class=\\"react-select__input\\"
                         style=\\"display: inline-block;\\"
                    >
                      <input autocapitalize=\\"none\\"
                             autocomplete=\\"off\\"
                             autocorrect=\\"off\\"
                             id=\\"react-select-configuration-sort-order-input\\"
                             spellcheck=\\"false\\"
                             tabindex=\\"0\\"
                             type=\\"text\\"
                             aria-autocomplete=\\"list\\"
                             style=\\"box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;\\"
                             value
                      >
                      <div style=\\"position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;\\">
                      </div>
                    </div>
                  </div>
                </div>
                <div class=\\"react-select__indicators css-mik995\\">
                  <span class=\\"react-select__indicator-separator css-1okebmr-indicatorSeparator\\">
                  </span>
                  <div aria-hidden=\\"true\\"
                       class=\\"react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer\\"
                  >
                    <svg height=\\"20\\"
                         width=\\"20\\"
                         viewbox=\\"0 0 20 20\\"
                         aria-hidden=\\"true\\"
                         focusable=\\"false\\"
                         class=\\"css-19bqh2r\\"
                    >
                      <path d=\\"M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z\\">
                      </path>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class=\\"mt-5\\">
      </div>
      <div class=\\"form-group\\">
        <div class=\\"text-center\\">
          <label class=\\"mb-4 font-weight-bold\\">
            Top labels threshold
          </label>
        </div>
        <div class=\\"mb-4\\">
          <div aria-disabled=\\"false\\"
               class=\\"input-range\\"
          >
            <span class=\\"input-range__label input-range__label--min\\">
              <span class=\\"input-range__label-container\\">
                0%
              </span>
            </span>
            <div class=\\"input-range__track input-range__track--background\\">
              <div style=\\"left: 0%; width: 33.33333333333333%;\\"
                   class=\\"input-range__track input-range__track--active\\"
              >
              </div>
              <span class=\\"input-range__slider-container\\"
                    style=\\"position: absolute; left: 33.33333333333333%;\\"
              >
                <span class=\\"input-range__label input-range__label--value\\">
                  <span class=\\"input-range__label-container\\">
                    33%
                  </span>
                </span>
                <div aria-valuemax=\\"99\\"
                     aria-valuemin=\\"0\\"
                     aria-valuenow=\\"33\\"
                     class=\\"input-range__slider\\"
                     draggable=\\"false\\"
                     role=\\"slider\\"
                     tabindex=\\"0\\"
                >
                </div>
              </span>
            </div>
            <span class=\\"input-range__label input-range__label--max\\">
              <span class=\\"input-range__label-container\\">
                99%
              </span>
            </span>
          </div>
        </div>
        <div class=\\"form-check form-check-inline\\">
          <span class=\\"custom-control custom-switch\\">
            <input id=\\"topLabelsConfigShow\\"
                   class=\\"custom-control-input\\"
                   type=\\"checkbox\\"
                   value
                   checked
            >
            <label class=\\"custom-control-label cursor-pointer mr-3\\"
                   for=\\"topLabelsConfigShow\\"
            >
              Show top labels
            </label>
          </span>
        </div>
      </div>
    </form>
  </div>
  <div class=\\"modal-footer\\">
    <span class=\\"text-muted\\">
      Version: unknown
    </span>
  </div>
</span>
"
`;
